auth/views.py

```diff
@@ -31,5 +36,47 @@ def braintree(request):
     return send(new_request)
 
 
+def braintree_verify(request):
+    """
+    Process Braintree webhooks. This is a pretty simple endpoint that
+    assumes the data has already been sanitised and cleaned by the solitude
+    server.
+
+    :bt_challenge string: the bt_challenge param sent by Braintree.
+
+    :status 200: contains the string to pass back.
+
+    Any other status is a failure to verify.
+    """
+    data = request.GET.get('bt_challenge')
+    res = WebhookNotificationGateway(BraintreeGateway()).verify(data)
+    return HttpResponse(res)
+
+
+def braintree_parse(request):
+    """
+    Validate Braintree webhook data. This assumes that solitude has cleaned
+    the data. It doesn't actually return any data
+
+    :bt_payload string: the payload from braintree.
+    :bt_signature string: the signature that needs to be verified.
+
+    :status 204: the data is to be trusted. Contains no content.
+
+    Any other status is a failure to verify and should not be trusted.
+    """
+    data = (
+        request.POST.get('bt_signature', ''),
+        request.POST.get('bt_payload', '')
```

Comment:
Well they will be unicode because Django. If the library doesn't complain then I'm happy :)